src/main/route/UserRouter.ts

```diff
@@ -1,34 +1,66 @@
 import { Router } from 'express';
 import { UserController } from '../controller/UserController';
+import { FirebaseAuth } from '../middleware/FirebaseAuth';
+import { auth } from 'firebase-admin/lib/auth';
+import Auth = auth.Auth;
 
 export class UserRouter {
   private userRouter = Router();
   private userController;
+  private firebaseAuth: FirebaseAuth;
 
-  constructor(usercontroller: UserController) {
-    this.userController = usercontroller;
+  constructor(userController: UserController, firebaseAdmin: Auth) {
+    this.userController = userController;
+    this.firebaseAuth = new FirebaseAuth(firebaseAdmin);
   }
 
   getRoutes(): Router {
     // profile endpoint
-    this.userRouter.get('/', this.userController.listAll);
-    this.userRouter.get('/:id', this.userController.getOneById);
-    this.userRouter.post('/', this.userController.newUser);
-    this.userRouter.patch('/:id', this.userController.editUser);
-    this.userRouter.delete('/:id', this.userController.deleteUser);
+    this.userRouter.get(
+      '/',
+      this.firebaseAuth.isAuthorizedForProtection,
+      this.userController.listAll
+    );
```

Comment:
Will we add middleware to `PromotionRouter`? We may want to authenticate the user for endpoints like `POST promotions` for example

Reply:
Yeah I think this is a good idea too, probably something you can do in another PR!

Reply:
We will need to discuss which endpoint we have to set the middleware.